### PR TITLE
Note for ODC new features and enhancements, Known issues and Bug fixes

### PR DESCRIPTION
**Pull request by Preeticp**
- This PR captures a note for New features and enhancement in Developer Perspective and the Known Issues and Bug Fixes for 4.5.
- This has been reviewed by the QE and SMEs

### CODE REVIEW

**Comment by Preeticp** [comment]
@nkakkar81 request a peer review for this PR please.

File: release_notes/ocp-4-5-release-notes.adoc [hunk 1]
```diff
@@ -193,6 +193,19 @@ You can now filter Operators by *Infrastructure Features* in OperatorHub. For
...
+[id="ocp-4-5-developer-perspective"]
+==== Developer Perspective
+
+You can now use the *Developer* perspective to:
+
+* Make informed decisions on installing Helm Charts in the *Developer Catalog* using the description and docs for them.
+* Uninstall, upgrade, and rollback Helm Releases.
+* Create and delete dynamic Knative event sources.
+* Deploy virtual machines, launch applications in them, or delete the virtual machines.
+* Provide Git webhooks, triggers, and  workspaces; manage credentials of private git repositories; and trouble shoot using better logs for Pipelines.
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:205]: s/triggers/Triggers
s/workspace/Workspace
s/trouble shoot/troubleshoot
s/Pipelines/the OpenShift Pipelines

File: release_notes/ocp-4-5-release-notes.adoc [hunk 2]
```diff
@@ -193,6 +193,19 @@ You can now filter Operators by *Infrastructure Features* in OperatorHub. For
...
+* Add health checks while deploying an application or after deploying it
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:206]: Add health checks during or after application deployment.

File: release_notes/ocp-4-5-release-notes.adoc [hunk 3]
```diff
@@ -557,6 +570,19 @@ for more details.
 [id="ocp-4-5-bug-fixes"]
 == Bug fixes
 
+* Previously, it was difficult to see the list of pods or resources associated with a Knative service in the *Topology* view. With this bug fix, when you select the Knative service, the sidebar displays a  list of pods along with a link to view the logs. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1801752[*BZ#1801752*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:573]: s/to see/to view
s/pods/Pods ( this is for entire doc)

File: release_notes/ocp-4-5-release-notes.adoc [hunk 4]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* When you edited an existing query using the promQl editor in the *metrics* tab of the *Monitoring* view, the cursor moved to the end of the line. With this bug fix, the promQl editor works as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1806114[*BZ#1806114*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:574]: s/When you/Previously, when you?
s/promQl - can be in backtics?

File: release_notes/ocp-4-5-release-notes.adoc [hunk 5]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* For Knative images, in the *Add* > *From Git* option, the *Advanced Options* for *Routing* would not provide a prefetched container port option. Also, if you created the service without updating the default port value of `8080`, the revisions would not show. With this bug fix, the user can select from the available port options using the drop-down list or provide input if they want to use another port and the revisions are shown as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1806552[*BZ#1806552*])
+* Previously, a Knative service created using the CLI could not be edited using the console as the images could not be fetched. Now, if the associated ImageStreams are not found while editing, the value provided by the user for the container image in the YAML file is used, which allows the user to edit the service using the console, even if the service was created using the CLI. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1806994[*BZ#1806994*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:576]: s/as the/because
s/YAML file is used. This allows

File: release_notes/ocp-4-5-release-notes.adoc [hunk 6]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* In the *Topology* view, editing the image name in the external image registry for a Knative service did not create a new revision. With this bug fix, a new revision of the service is created when the name of the service is changed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1807868[*BZ#1807868*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:577]: s/In the/Previously, in

File: release_notes/ocp-4-5-release-notes.adoc [hunk 7]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* When you used the *Add* > *Container Image* option, and then selected the *Image stream tag from internal registry* option, the *ImageStreams* drop-down list did not list the option to deploy images from the *OpenShift* namespace. However, the user could access them through the CLI. With this bug fix, all users have access to images in the *OpenShift* namespace through the console and the CLI. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1822112[*BZ#1822112*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:578]: s/the user could/the user was able to

File: release_notes/ocp-4-5-release-notes.adoc [hunk 8]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* Previously, in the *Pipeline Builder*, when you edited a Pipeline that referenced a Task that did not exist, the entire screen would go white. Now you see an icon to indicate that an action is required and a drop-down list is displayed to easily update the Task reference. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1839883[*BZ#1839883*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:579]: s/Now you see an icon to indicate that an action is required and a drop-down list is displayed to easily update the Task reference./ This fix displays an icon that ...

File: release_notes/ocp-4-5-release-notes.adoc (hunk 6 above)
Comment (Preeticp) [reply release_notes/ocp-4-5-release-notes.adoc:577]: Trying to avoid using Previously too many times.

File: release_notes/ocp-4-5-release-notes.adoc [hunk 9]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* In the *Pipelines Details* page, when you changed existing fields in the *Parameters* and the *Resources* tabs, the *Save* button was disabled though the new changes were detected. The validation criteria has now been modified and the *Save* button is enabled to submit changes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1804852[*BZ#1804852*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:580]: s/was disabled though/was disabled even though?

File: release_notes/ocp-4-5-release-notes.adoc [hunk 10]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* In the *Add* > *From Git* option, the pipeline templates provided by the OpenShift Pipelines Operator would fail when the *Deployment* or *Knative Services* resource options were selected. This bug fix adds support to use the resource type as well as the runtime to determine the pipeline template, thus providing resource-specific Pipeline templates. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1796185[*BZ#1796185*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:581]: s/pipeline template /Pipeline template? ( 2 places)

File: release_notes/ocp-4-5-release-notes.adoc [hunk 11]
```diff
@@ -557,6 +570,19 @@ for more details.
...
+* When a Pipeline is created using the *Pipeline Builder* and a Task parameter of the type array is used, the Pipeline does not start. With this bug fix, both array and string type parameters are supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1813707[*BZ#1813707*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:582]: s/and a Task  parameter of the type array is used/ , with a Task parameter of type array ?

File: release_notes/ocp-4-5-release-notes.adoc [hunk 12]
```diff
@@ -733,6 +759,13 @@ the imagestream has the pull-through policy enabled. In this case, an
 `ImagePullBackOff` error displays.
 (link:https://bugzilla.redhat.com/show_bug.cgi?id=1787112[*BZ#1787112*])
 
+* The Che workspace Operator was updated to remove support for the Workspace custom resource and add support for the DevWorkspace custom resource. However, the OpenShift web terminal continues to use the Workspace custom resource instead of the DevWorkspace custom resource for starting the terminal workspaces. Hence it fails to work with the latest version of the Che workspace Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1846969[*BZ#1846969*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:762]: s/workspace/Workspace?

File: release_notes/ocp-4-5-release-notes.adoc [hunk 13]
```diff
@@ -733,6 +759,13 @@ the imagestream has the pull-through policy enabled. In this case, an
...
+
+* A basic-user is unable to view the *Dashboard* and *Metrics* tabs in the *Monitoring* view of the *Developer* perspective. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1846409[*BZ#1846409*])
```
Comment (nkakkar81) [inline release_notes/ocp-4-5-release-notes.adoc:764]: basic-user in backtics?

File: release_notes/ocp-4-5-release-notes.adoc [hunk 14]
```diff
@@ -557,6 +570,19 @@ for more details.
 [id="ocp-4-5-bug-fixes"]
 == Bug fixes
 
+* Previously, it was difficult to see the list of Pods or resources associated with a Knative service in the *Topology* view. With this bug fix, when you select the Knative service, the sidebar displays a list of Pods along with a link to see the logs. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1801752[*BZ#1801752*])
+* When you edited an existing query using the promQl editor in the *metrics* tab of the *Monitoring* view, the cursor moved to the end of the line. With this bug fix, the promQl editor works as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1806114[*BZ#1806114*])
```
Comment (lamek) [inline release_notes/ocp-4-5-release-notes.adoc:574]: promQL (The "L" should be capitalized, I think)

File: release_notes/ocp-4-5-release-notes.adoc [hunk 15]
```diff
@@ -733,6 +759,13 @@ the imagestream has the pull-through policy enabled. In this case, an
 `ImagePullBackOff` error displays.
 (link:https://bugzilla.redhat.com/show_bug.cgi?id=1787112[*BZ#1787112*])
 
+* The Che Workspace Operator was updated to remove support for the Workspace custom resource and add support for the DevWorkspace custom resource. However, the OpenShift web terminal continues to use the Workspace custom resource instead of the DevWorkspace custom resource for starting the terminal workspaces. Hence it fails to work with the latest version of the Che Workspace Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1846969[*BZ#1846969*])
```
Comment (lamek) [inline release_notes/ocp-4-5-release-notes.adoc:762]: This one was a little confusing to me. I suggest rewording this. Here is one possibility:

"The Che Workspace Operator was updated to use the DevWorkspace custome resource instead of the Workspace custom resource. However, the OpenShift web terminal continues to use the Workspace custom resource. Because of this, the OpenShift web terminal fails to work with the latest version of the Che Workspace Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1846969[*BZ#1846969*])"

**Review by lamek — COMMENTED** [review]
A few comments and otherwise LGTM.

File: release_notes/ocp-4-5-release-notes.adoc (hunk 14 above)
Comment (Preeticp) [reply release_notes/ocp-4-5-release-notes.adoc:574]: Apparently the P should be capitalized too

Comment (Preeticp) [reply release_notes/ocp-4-5-release-notes.adoc]: Fixed

File: release_notes/ocp-4-5-release-notes.adoc (hunk 15 above)
Comment (Preeticp) [reply release_notes/ocp-4-5-release-notes.adoc]: Fixed

**Comment by Preeticp** [comment]
@lamek I have fixed the comments and squashed the commits. Can you please merge? This has been created on the 4.5 branch itself.
